packages/fetchai/skills/carpark_client/handlers.py

```diff
@@ -19,15 +19,145 @@
+from packages.fetchai.protocols.oef_search.message import OefSearchMessage
+from packages.fetchai.skills.generic_buyer.dialogues import (
+    FipaDialogue,
+    FipaDialogues,
+    OefSearchDialogue,
+)
 from packages.fetchai.skills.generic_buyer.handlers import (
     GenericFipaHandler,
     GenericLedgerApiHandler,
     GenericOefSearchHandler,
     GenericSigningHandler,
 )
+from packages.fetchai.skills.generic_buyer.strategy import GenericStrategy
+from packages.fetchai.skills.carpark_client.strategy import Strategy
 
-
-FipaHandler = GenericFipaHandler
 LedgerApiHandler = GenericLedgerApiHandler
-OefSearchHandler = GenericOefSearchHandler
 SigningHandler = GenericSigningHandler
+
+
+class FipaHandler(GenericFipaHandler):
+    """This class handles fipa messages."""
+
+    def __init__(self, **kwargs: Any) -> None:
+        super().__init__(**kwargs)
+        self.context.carpark_data = cast(list, [])
+
+    def _handle_propose(
+        self, fipa_msg: FipaMessage, fipa_dialogue: FipaDialogue
+    ) -> None:
+        """
+        Handle the propose.
+
+        :param fipa_msg: the message
+        :param fipa_dialogue: the dialogue object
+        """
+        sender = fipa_msg.sender[-5:]
+        self.context.logger.info(
+            f"received proposal={fipa_msg.proposal.values} from sender={sender}"
+        )
+        strategy = cast(Strategy, self.context.strategy)
+        self.context.carpark_data.append(
+            {
+                "sender": fipa_msg.sender,
+                "message": fipa_msg,
+                "decision": FipaMessage.Performative.DECLINE
+                if not strategy.is_acceptable_proposal(fipa_msg.proposal)
+                or not strategy.is_affordable_proposal(fipa_msg.proposal)
+                else None,
+            }
+        )
+        # self.context.logger.info(
+        #     f"fipa_message_id={fipa_msg.message_id} with target={fipa_msg.target} added to carpark_data."
+        # )
+
+        # makes more sense to only wait for a certain time
+        if len(self.context.carpark_data) == len(self.context.carpark_agents):
+            self.context.logger.info("received all proposals, making decision...")
+            undecided_proposals = list(
+                filter(lambda x: x["decision"] is None, self.context.carpark_data)
+            )
+            cheapest_proposal = strategy.get_cheapest_proposal(undecided_proposals)
+            for carpark in undecided_proposals:
+                cheapest = cheapest_proposal["sender"] == carpark["sender"]
+                carpark["decision"] = (
+                    FipaMessage.Performative.ACCEPT
+                    if cheapest
+                    else FipaMessage.Performative.DECLINE
+                )
+            self._handle_propose_send(strategy)
+
+    def _handle_propose_send(self, strategy: GenericStrategy) -> None:
+        """
+        The actual sending of the accept/decline messages.
+
+        :param strategy: _description_
```

Comment:
I guess `_description_` was meant to be replaced.

Reply:
fixed already, I simply forgot to add something